### PR TITLE
Add drop monitor Kernel Patches for buffer support

### PR DESCRIPTION
**Pull request by vivekrnv**
Add the new patch 0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch and move the patch names under the markers 

 ## Patch List
* 0001-psample-Encapsulate-packet-metadata-in-a-struct.patch : https://github.com/torvalds/linux/commit/a03e99d39f1943ec88f6fd3b0b9f34c20663d401
* 0002-psample-Add-additional-metadata-attributes.patch : https://github.com/torvalds/linux/commit/07e1a5809b595df6e125504dff6245cb2c8ed3de
* 0003-psample-define-the-macro-PSAMPLE_MD_EXTENDED_ATTR.patch :
* 0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch :

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

**Review by dgsudharsan — APPROVED** [review]

File: patch/0003-psample-define-the-macro-PSAMPLE_MD_EXTENDED_ATTR.patch [hunk 1]
```diff
@@ -1,28 +1,13 @@
...
-Subject: [PATCH 3/3] psample: Add Nvidia-specific wrapper function for the psample driver
-
-Add a variable to notify the psample driver to use a wrapper function,
-which does preprocess before sending the sample packet to the userspace application.
-
-Signed-off-by: Vadym Hlushko <vadymh@nvidia.com>
----
- include/net/psample.h | 2 ++
- 1 file changed, 2 insertions(+)
-
```
Comment (saiarcot895) [inline patch/0003-psample-define-the-macro-PSAMPLE_MD_EXTENDED_ATTR.patch:13]: Can this header be preserved?

File: patch/0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch [hunk 2]
```diff
@@ -0,0 +1,230 @@
+From 1eee33e3c2c47fe2ca9dbad0712c0686ecdbf60a Mon Sep 17 00:00:00 2001
+From: Zhengfeng Dou <zhengfengd@nvidia.com>
+Date: Thu, 29 Jun 2023 12:15:19 +0800
+Subject: [PATCH] drop_monitor: Extend WJH buffer linux channel
+
```
Comment (saiarcot895) [inline patch/0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch:5]: Can you add a description here?

File: patch/0003-psample-define-the-macro-PSAMPLE_MD_EXTENDED_ATTR.patch (hunk 1 above)
Comment (vivekrnv) [reply patch/0003-psample-define-the-macro-PSAMPLE_MD_EXTENDED_ATTR.patch:13]: Updated

File: patch/0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch (hunk 2 above)
Comment (vivekrnv) [reply patch/0004-drop_monitor-Extend-WJH-buffer-linux-channel.patch:5]: Updated

File: patch/series [hunk 3]
```diff
@@ -193,9 +197,6 @@ cisco-hwmon-pmbus_core-pec-support-check.patch
 0001-hwmon-emc2305-Fix-unable-to-probe-emc2301-2-3.patch
 
 # sFlow + dropmon support
```
Comment (saiarcot895) [inline patch/series:199]: Can you remove this line, since the patches have been moved up?

Comment (vivekrnv) [reply patch/series:199]: Sure, will do

**Review by saiarcot895 — APPROVED** [review]

**Review by paulmenzel — CHANGES_REQUESTED** [review]
Thank you for the patches. Two things:

1.  Could you please comment, when patches 3 and 4 are going to be upstreamed?
2.  Could you please rebase the changes with only three commits: recreate patches 1 and 2, move the patches for the marker, and add patch 4?

**Comment by vivekrnv** [comment]
> Thank you for the patches. Two things:
> 
> 1. Could you please comment, when patches 3 and 4 are going to be upstreamed?
> 2. Could you please rebase the changes with only three commits: recreate patches 1 and 2, move the patches for the marker, and add patch 4?

They are downstream patches and won't be upstreamed. The patches 0003 & 0004 offer logically separate functionality and so i don't think it is a good idea to merge 0003 & 0004 together

**Pull request merged**